src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/NodeMonitoringConnectionPlugin.java

```diff
@@ -84,9 +85,10 @@ public NodeMonitoringConnectionPlugin(
       ICurrentConnectionProvider currentConnectionProvider,
       PropertySet propertySet,
       IConnectionPlugin nextPlugin,
-      Log logger) {
+      Log logger) throws SQLException {
```

Comment:
It seems that the only source of SQLException in the constructor body is generateNodeKeys(). I'd suggest to remove that call and relocate it to later stages of plugin lifecycle, like to execute().

Reply:
Sounds good